file.go

```diff
@@ -0,0 +1,136 @@
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"fmt"
+	"net/http"
+	"io"
+	"path/filepath"
+	"bytes"
+	"archive/zip"
+	"strings"
+)
+
+// Download the zip file at the given URL to a temporary local directory.
+// Returns the asbolute path to the downloaded zip file.
+// IMPORTANT: You must call defer os.RemoveAll(dir) when done with this directory!
+func downloadGithubZipFile(repoOwner, repoName, gitTag, githubToken string) (string, *fetchError) {
+	// Create a temp directory
+	tempDir, err := ioutil.TempDir("", "")
+	if err != nil {
+		return "", newErr(err)
+	}
+
+	// Create an empty file to write to
+	file, err := os.Create(filepath.Join(tempDir, "repo.zip"))
+	if err != nil {
+		return "", newErr(err)
+	}
+	defer file.Close()
+
+	// Define the url
+	url := fmt.Sprintf("https://api.github.com/repos/%s/%s/zipball/%s", repoOwner, repoName, gitTag)
```

Comment:
The `repoOwner`, `repoName`, and `gitTag` params may need URL encoding. Does golang have a library for working with URLs?